File: src/CScourse/arrayChunking.js
// Given an array and a Chunk size, 
// divide the array into many subarrays where 
// each subarray is of the provided size

function chunk(array, size) {
    let result = []
    for (arr of array) {
        let prev = result[result.length - 1]
        if (!prev || prev.length==size) {
            result.push([arr])
        } else {
            result[result.length - 1].push(arr)
        }
    }
    return result
}

function chunkSlice(array, size) {
        let result = []
        let loops = parseInt(array.length / size)
        if (array.length%size !== 0) loops++
        for(let i = 0; i<loops; i++) {
            result.push(array.slice(i*size, (size+(size*i))))
        }
        return result
}

File: src/CScourse/ArrayChunking.rb
# Given an array and a Chunk size, 
# divide the array into many subarrays where 
# each subarray is of the provided size

def array_chunk(array, size)
    result = []
    index = 0
    while array.length > index do
        result.push(array.slice(index, index + size)) # see below for difference with .slice in Ruuby and JavaScript
        index += size
    end
    return result
end

# .slice in JavaScript = .slice(indexToBeginAt, indexToEndAt - 1)
# .slice in Ruby = .slice(index_to_begin_at, number of elements to return)